src/main/groovy/life/qbic/samplestatus/reporter/Result.groovy

```diff
@@ -0,0 +1,137 @@
+ * same instance.</p>
+ *
+ * To properly deal with Result objects, a good idiom looks like this:
+ *
+ * <pre>
+ *
+ * final Result\\<String, Exception\> result = new Result("Contains actual information")
+ * // access the value
+ * switch (result) {
+ *    case { it.isOk() }:
+ *      doSomething(it.getValue())
+ *      break
+ *    case { it.isError() }:
+ *      doErrorHandling(it.getError())
+ *      break
+ * }
+ * </pre>
+ *
+ * @since 0.1.0
+ */
+class Result<V, E extends Exception> {
+
+    private final V value
+    private final E err
+
+    /**
+     * Static constructor method for creating a result object instance of type <code>V,E</code>
+     * wrapping an actual value <code>V</code>.
+     * @param value the notorious value to get wrapped in a result object
+     * @return a new result object instance
+     */
+    static <V,E extends Exception> Result<V, E> of(V value) {
+        return new Result<>(value)
+    }
+
+    /**
+     * Static constructor method for creating a result object instance of type <code>V,E</code>
+     * wrapping an error <code>E</code>.
+     * @param e the suspicious error to get wrapped in a result object
+     * @return a new result object instance
+     */
+    static <V,E extends Exception> Result<V, E> of(E e) {
+        return new Result<>(e)
+    }
+
+    private Result(V value) {
+        this.value = value
+        this.err = null
+    }
+
+    private Result(E exception) {
+        this.value = null
+        this.err = exception
+    }
+
+    /**
+     * Access the wrapped value if present
+     * @return the wrapped value
+     * @throws NoSuchElementException if no value exists in the result object
+     */
+    V getValue() throws NoSuchElementException {
+        if (!value) {
+            throw new NoSuchElementException("Result with error has no value.")
+        }
+        return value
+    }
+
+    /**
+     * Access the wrapped error if present
+     * @return the wrapped exception
+     * @throws NoSuchElementException if no error exists in the result object
+     */
+    E getError() throws NoSuchElementException {
+        if (!err) {
+            throw new NoSuchElementException("Result with value has no error.")
+        }
+        return err as E
+    }
+
+    /**
+     * Returns <code>true</code>, if the result object contains an error. Is always the negation
+     * of {@link Result#isOk()}.
+     * <p>So <code>{@link Result#isError()} == !{@link Result#isOk()}</code></p>
+     * @return true, if the result object has an error, else false
+     */
+    Boolean isError() {
+        return err as boolean
+    }
+
+    /**
+     * Returns <code>true</code>, if the result object contains an error. Is always the negation
+     * of {@link Result#isError()}.
+     * <p>So <code>{@link Result#isOk()} == !{@link Result#isError()}</code></p>
+     * @return true, if the result object has a value, else false
+    */
+    Boolean isOk() {
+        return value as boolean
```

Comment:
As this is always the negation as stated in the JavaDoc, we could make it clear in the code as well.
```suggestion
        return value as boolean && !isError()
```

Reply:
Ok, but I argue, that this is not possible (have a look in the static constructor methods). This state cannot happen.

Reply:
Yes, this check does not change the output now. What happens if a new constructor is added? I think this makes the behaviour clearer. Another option would be to add tests for this behaviour or to remove the assumption from the documentation (I like the tests)